### PR TITLE
BUGFIX for ticket #581

### PR DESCRIPTION
**Pull request by Liametc**
### Added

- Added NODE_CATEGORY to all node handlers so doesn't pick up wrong settings when duplicate node types exist.
- Added 2 new classes to accommodate this new feature as can't use other classes.

### Fixed 

- Fixed issue with loader dialog not picking up correct publish types.

### Changed

- Changed Single Frame to Single File on param label.

### CODE REVIEW

File: python/tk_houdini/base_hooks/node_handler_base.py [hunk 1]
```diff
@@ -38,13 +39,17 @@ def __new__(cls, *args, **kwargs):
...
                 return super(NodeHandlerBase, cls).__new__(cls, *args, **kwargs)
             engine = sgtk.platform.current_engine()
             node_handlers = engine.get_setting("node_handlers")
             for handler in node_handlers:
-                if handler["node_type"] == cls.NODE_TYPE:
+                conditions = [
+                    handler["node_type"] == cls.NODE_TYPE,
+                    handler["node_category"] == cls.NODE_CATEGORY
+                ]
+                if all(conditions) :
```
Comment (j0yu) [inline python/tk_houdini/base_hooks/node_handler_base.py]:
```suggestion
                matching_node_category_type = (
                    handler["node_type"] == cls.NODE_TYPE
                    and handler["node_category"] == cls.NODE_CATEGORY
                )
                if matching_node_category_type :
```

Constructing a new list of booleans seems excessive

File: hooks/node_handlers/backgroundimageplane_handler.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Designed to inherit ``base_import_handler`` i.e. define the ``hook`` attribute
+in your project configuration settings with the following inheritance:
+
+.. code-block:: yaml
+    :caption: env/includes/settings/tk-houdini_node_handlers.yml
+
+    node_handlers.asset_step:
+    - node_type: file
+      node_category: Cop2
```
Comment (j0yu) [inline hooks/node_handlers/backgroundimageplane_handler.py]:
```suggestion
    - node_type: backgroundimageplane
      node_category: Object
```

Should these match the class attributes?

File: hooks/node_handlers/alembicarchive_import_handler.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Designed to inherit ``base_import_handler`` i.e. define the ``hook`` attribute
+in your project configuration settings with the following inheritance:
+
+
+.. code-block:: yaml
+    :caption: env/includes/settings/tk-houdini_node_handlers.yml
+
+    node_handlers.shot_step:
+    - node_type: alembic
+      node_category: Sop
```
Comment (j0yu) [inline hooks/node_handlers/alembicarchive_import_handler.py]:
```suggestion
    - node_type: alembicarchive
      node_category: Object
```

Comment (j0yu) [reply hooks/node_handlers/alembicarchive_import_handler.py]: Should these match the class attributes?

**Review by j0yu — CHANGES_REQUESTED** [review]

File: hooks/node_handlers/backgroundimageplane_handler.py (hunk 2 above)
Comment (Liametc) [reply hooks/node_handlers/backgroundimageplane_handler.py:11]: weird, this is not what my vscode is showing me

File: hooks/node_handlers/alembicarchive_import_handler.py (hunk 3 above)
Comment (Liametc) [reply hooks/node_handlers/alembicarchive_import_handler.py:12]: also this

**Review by j0yu — APPROVED** [review]

**Pull request merged**